New Proposal: Traders (#557)
space-wizards/docs: src/en/space-station-14/round-flow/proposals/traders-ghost-role.md

## Overview
<!-- A very short, maybe three sentence summary of what this proposal is about. A high level "overview" or "what this adds". -->
The trader is a mid-round ghost role that is largely crew aligned. Each trader is given a unique shuttle (like the visitor) that is equipped to function as a mobile shop. Traders get a randomized inventory of items, and in order to access these items, an item must be placed into the trader's vault. Traders get an open-ended objective to make as much profit as possible without getting banned from the station, and must haggle, barter, and price on their own terms to fulfill this goal.

## Background
<!--
Summarize any information that is needed to contextualize the proposed changes, e.g. the current state of the game.
Also link any relevant discussions on Discord, GitHub, or HackMD that are relevant to the proposal.
-->
Every time someone goes to the lengths to set up a shop in SS14, everyone else seems to love it from both a mechanical and roleplay sense. Having done so myself, haggling with people over exchange prices or spesos and defending your shop from threats is probably the most fun thing I have done in this game.

> In addition, at the moment there is a lack of crew-affiliated ghost roles that encourage unique gameplay. Visitors are pretty much just a way to respawn. Station pets don't do much until you cognizine them. Mothroaches and mice are glorified spectators with food mechanics. Derelict borgs are probably the most interesting ghost role, but they are apparently just a bit too common at the moment.

Reviewer: ```suggestion
Every time someone goes to the lengths to set up a shop in SS14, everyone else seems to love it from both a mechanical and roleplay sense. Having done so myself, haggling with people over exchange prices or spesos and defending your shop from threats is probably the most fun thing I have done in this game.

In addition, at the moment, there is a lack of crew-affiliated ghost roles that encourage unique gameplay. Visitors are pretty much just a way to respawn. Station pets don't do much until you cognizine them. Mothroaches and mice are glorified spectators with food mechanics. Derelict borgs are probably the most interesting ghost role, but they are apparently just a bit too common at the moment.
```

## Features
<!-- Give a description of what game mechanics you would like to add or change. This should be a general overview, with enough details on critical design points that someone can directly implement the feature from this design document. Exact numbers for game balance however are not necessary, as these can be adjusted later either during development or after it has been implemented, but mention *what* will have to be balanced and what needs to be considered when doing so. -->
Traders spawn on a trader shuttle, with some new machines that help them handle transactions and protect both their inventory and their spoils. These machines only allow access to people with the trader's specific DNA, meaning the trader does not have to worry about getting robbed blind during a snack or drink break. When a trader spawns, an announcement is given that a trader is approaching the station.

Traders have in-built flash protection, to prevent them from ever becoming a revolutionary.

### Machines:

Bluespace Vault:

> A large, indestructible machine that cannot be manually unanchored. All of the items that the trader takes as payment end up here, and traders can freely retreive and deposit items from it. Traders can also add items in the vault into their lockboxes' inventories, recycling payment into product. The only items this cannot store are "trash" items (whatever can go in a trash bag) and objective items.

Reviewer: ```suggestion
A large, indestructible machine that cannot be manually unanchored. All items the trader accepts as payment end up here, and traders can freely retrieve and deposit items from it. Traders can also add items from the vault to their lockboxes' inventories, recycling payments into products. The only items this cannot store are "trash" items (whatever can go in a trash bag) and objective items.
```

Bluespace Lockbox:
Multiple indestructible machines that cannot be manually unanchored. Each contains a different "category" of items that can be bought from the trader, such as clothes, materials, weaponry, etc., and are filled up with a random selection of those items when the trader spawns. In order for an item to be dispensed, an item must first be put into the lockbox, and that item will be sent into the trader's vault when the item is selected. Lockboxes have a limit to how much they can stock, and if a trader wants to refill them back up to this limit, they have to use the Restock Radio they spawn with while on the Automated Trade Station.

Traders can select an item from the lockbox's inventory to be displayed above it as a hologram. This allows them to showcase their stock without compromising its safety.

Bluespace Cart:

> A draggable item that can be linked to a lockbox with a network configurator to preform that lockbox's function on the go. Useful if the trader can't find a good place to dock with the station, but can only be linked to one lockbox at a time. Convenience of location at the expense of your selection.

Reviewer: ```suggestion
A draggable item that can be linked to a lockbox with a network configurator to perform that lockbox's function on the go. Useful if the trader can't find a good place to dock with the station, but can only be linked to one lockbox at a time. Convenience of location at the expense of your selection.
```

Item Categories:
Clothing (Costume pieces and rare clothing sets.)
Raw Materials (Low amounts of these, so as not to step on cargo's toes. But if the salvs are slacking...)
Personal Defense (Low-powered guns, ammo, and some melee weapons. Lets Security get in on the fun.)
Protective Clothing (Hardsuits and other clothing items with unique abilities)
Tech (Mostly things that science can research, but potentially a bit earlier than usual.)
Food (Mostly ingredients to allow the chef to make some new dishes, with the odd Arnold's pizza.)
Experimental (Implants, other weird stuff.)

> Traders only have four lockboxes, meaning that some of the categories will not be available for purchase that round.

Reviewer: ```suggestion
Item Categories:
Clothing (Costume pieces and rare clothing sets.)
Raw Materials (Low amounts of these, so as not to step on cargo's toes. But if the salvs are slacking...)
Personal Defense (Low-powered guns, ammo, and some melee weapons. Let's Security get in on the fun.)
Protective Clothing (Hardsuits and other clothing items with unique abilities)
Tech (Mostly things that science can research, but potentially a bit earlier than usual.)
Food (Mostly ingredients to allow the chef to make some new dishes, with the odd Arnold's pizza.)
Experimental (Implants, other weird stuff.)

Traders have only four lockboxes, so some categories will not be available for purchase that round.
```

There is exactly one specific interaction traders have with antagonists: their machines can be broken into with an Authentication Disrupter, which gives everyone access to their contents.

### Optional Feature: Brands
In this feature, shuttles would have identifiable branding, and the traders might even be locked to a specific species dependent on the brand. In addition, one of the categories would stocked with items unique to that brand. 

### Optional Feature: Contraband
In this feature, traders would occasionally have contraband items distributed in with their stock, such as having a single item of chameleon clothing in their clothing lockbox, or perhaps a hypopen in their experimental lockbox.

Although they are free to sell this to anyone aboard the station, since they are not crew, they can potentially be banned from selling at the station if they make too many risky deals. So, there's nothing mechanically keeping you from selling that EMAG to a scientist head revolutionary ... but if that revolution goes south and your deal gets traced, well, you're now a criminal.

### Optional Feature: Theivin' Traders

> In this feature, a separate ghost role for an antagonistic trader would sometimes appear. This trader would spawn with (item-only) theif objectives, the pickpocketing ability, and the pacifism implant. They would have to accomplish their objectives without any of the theif's original tools ... but with the full power of their shop's random inventory and their business skills. Every man has his price, right?

Reviewer: ```suggestion
### Optional Feature: Contraband
In this feature, traders would occasionally have contraband items distributed with their stock, such as a single item of chameleon clothing in their clothing lockbox or perhaps a hypopen in their experimental lockbox.

Although they are free to sell this to anyone aboard the station, since they are not crew, they can potentially be banned from selling at the station if they make too many risky deals. So, there's nothing mechanical keeping you from selling that EMAG to a scientist-head revolutionary... but if that revolution goes south and your deal gets traced, well, you're now a criminal.

### Optional Feature: Theivin' Traders
In this feature, a separate ghost role for an antagonistic trader would sometimes appear. This trader would spawn with (item-only) thief objectives, the pickpocketing ability, and the pacifism implant. They would have to accomplish their objectives without any of the thief's original tools ... but with the full power of their shop's random inventory and their business skills. Every man has his price, right?
```

## Administrative & Server Rule Impact (if applicable)
<!--
- Does this feature introduce any new rule enforcement challenges or additional workload for admins?
- Could this feature increase the likelihood of griefing, rule-breaking, or player disputes?
- How are the rules enforced mechanically by way the feature will be implemented?
-->
The most troublesome feature is giving traders contraband and/or weaponry, as they would be able to use it freely. However, traders should be rare enough, and randomized enough, that any griefing a trader could do would be contained to only about one round at most.

Mechanically, traders are a bit lax, except for having to put an item into their Vault to dispense something from their stock. However, this is important to allow for unique pricing on the part of each trader. 
## Technical Consideration
<!--
- Are there any anticipated performance impacts?
- Does the feature require new systems, UI elements, or refactors of existing ones?
- For required UI elements, give a short description or a mockup of how they should look like (for example a radial menu, actions & alerts, navmaps, or other window types)
-->

> New code would have to be written for the trader's unique machines, and each would need a UI (probably adapted from the menus for lathes or vendors) that shows their contents. In the lockboxes' cases, a section would need to be added that shows the item currently inserted as the price.

Reviewer: ```suggestion
New code would need to be written for the trader's unique machines, and each would need a UI (probably adapted from the lathes or vendor menus) that displays its contents. In the lockboxes' cases, a section would need to be added that shows the item currently inserted as the price.
```
